code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/volume/VolumeActionFilter.java

```diff
@@ -0,0 +1,44 @@
+package io.cattle.platform.iaas.api.filter.volume;
+
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.Mount;
+import io.cattle.platform.core.model.Volume;
+import io.github.ibuildthecloud.gdapi.model.Resource;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.response.ResourceOutputFilter;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class VolumeActionFilter implements ResourceOutputFilter {
+
+    @Inject
+    GenericMapDao mapDao;
+
+    @Override
+    public Resource filter(ApiRequest request, Object original, Resource converted) {
+        if (!(original instanceof Volume)) {
+            return converted;
+        }
+        Volume vol = (Volume)original;
+        List<? extends Mount> mounts = mapDao.findNonRemoved(Mount.class, Volume.class, vol.getId());
+        if (mounts.size() > 0) {
+            converted.getActions().remove("deactivate");
+            converted.getActions().remove("remove");
+            converted.getActions().remove("purge");
+        }
+        return converted;
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { Volume.class };
+    }
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { "volume" };
```

Comment:
The string type isn't required if it is the string equivalent of the getTypeClasses().  So you can just return empty array.